File: src/main/kotlin/ch/wngr/bookstore/models/SaleDTO.kt
package ch.wngr.bookstore.models

data class SaleDTO(
    val id: Int?,
    val isbn: String?,
    val title: String,
    val price: Float,
    val fullPrice: Float,
    val quantity: Int?,
    val new: Boolean,
)

data class SaleList(
    val sales: List<SaleDTO>,
    val paymentMethod: List<PaymentOptionDTO>?,
    val priceWithDiscount: Float,
    val priceWithoutDiscount: Float,
    val priceDiscountPercent: Int,
)

data class PaymentOptionDTO(
    val value: String,
    val price: Float,
)

File: src/frontend/src/models/SaleDto.ts
import { ScraperBook } from "@/models/ScraperBook";

export class SaleDto {
  id!: number | undefined;
  isbn: string | undefined;
  title!: string;
  price!: number;
  fullPrice!: number;
  quantity: number | undefined;
  new!: boolean;
  originalBook!: ScraperBook;

  public isEqual(sale: SaleDto) {
    return (
      this.id === sale.id &&
      this.isbn === sale.isbn &&
      this.title === sale.title &&
      this.price === sale.price &&
      this.fullPrice === sale.fullPrice &&
      this.quantity === sale.quantity &&
      this.new === sale.new
    );
  }
}

export class PaymentOptionDto {
  value!: string;
  price!: number;
}

export class SaleList {
  sales!: SaleDto[];
  paymentMethod!: PaymentOptionDto[];
  priceWithDiscount!: number;
  priceWithoutDiscount!: number;
  priceDiscountPercent!: number;
}
